example/conv2DWithMdspan/src/conv2DWithMdspan.cpp

```diff
@@ -0,0 +1,243 @@
+/* Copyright 2023 Mehmet Yusufoglu, Bernhard Manfred Gruber, René Widera
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+// Needed for running example for all backends available; one by one
+#include <alpaka/example/ExecuteForEachAccTag.hpp>
+
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+//! Convolution Example using Mdspan structure to pass multi-dimensional data to the kernel
+//!
+//! A 2D Convolutional filter applied to a matrix. Mdspan data structure is used, therefore pitch and size values are
+//! not needed to be passed to the kernel. Results can be tested by comparing with the results of the Matlab call: Y =
+//! filter2(FilterMatrix,InputMatrix,'same');
+
+/**
+ * @brief 2D Convolutional Filter using only global memory for the input-matrix and the filter-matrix
+ */
+struct ConvolutionKernelMdspan2D
+{
+    //! \tparam TAcc Accelerator type
+    //! \tparam TElem The input-matrix and filter-matrix element type
+    //! \param acc Accelerator
+    //! \param input Input matrix
+    //! \param output Output matrix
+    //! \param filter Filter-matrix
+
+    template<typename TAcc, typename MdSpan>
+    ALPAKA_FN_ACC auto operator()(TAcc const& acc, MdSpan const input, MdSpan output, MdSpan const filter) const
+        -> void
+    {
+        auto matrixWidth = input.extent(0);
+        auto matrixHeight = input.extent(1);
+        // Filter matrix is square
+        auto filterWidth = filter.extent(0);
+
+        // Get thread index, the center of filter-matrix is positioned to the item on this index.
+        auto const [row, col] = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+
+        // The convolutional filter-matrix applied to the input matrix, it's position is row and col. If some of the
+        // items of the filter are outside the matrix, those are not taken into calculation or they are assumed zero.
+        if(col < matrixWidth && row < matrixHeight)
+        {
+            float pValue{0.0f};
+            for(int32_t fRow = 0; fRow < static_cast<int32_t>(filterWidth); fRow++)
+            {
+                for(int32_t fCol = 0; fCol < static_cast<int32_t>(filterWidth); fCol++)
+                {
+                    // Position of input matrix element to be multiplied with the corresponding element at filter
+                    auto const exactRow = static_cast<int32_t>(row) - filterWidth / 2 + fRow;
+                    auto const exactCol = static_cast<int32_t>(col) - filterWidth / 2 + fCol;
+                    if(exactRow >= 0 && exactRow < matrixHeight && exactCol >= 0 && exactCol < matrixWidth)
+                    {
+                        pValue += filter(fRow, fCol) * input(exactRow, exactCol);
+                    }
+                }
+                output(row, col) = pValue;
```

Comment:
shouldn't this be outside the loop on `fRow` ?

Reply:
Yes, fixed.  (Good catch. It was not changing the result therefore can not be detected.)